### PR TITLE
Fixed #34180 -- Added note ref resetting language in test tear-downs.

### PR DESCRIPTION
**Pull request by fnaimi66**
Fixed #34180 -- Wrote addition to docs to address a bug.

Added an explanation about a bug that changes the active language in subsequent tests when the test language is by using a cookie or HTTP header

### CODE REVIEW

File: docs/topics/testing/tools.txt [hunk 1]
```diff
@@ -687,6 +687,13 @@ or by including the ``Accept-Language`` HTTP header in the request::
 
 More details are in :ref:`how-django-discovers-language-preference`.
 
+When setting the language for a test client request by either creating a cookie or by including the Accept-Language HTTP header, the Django active language will switch for all other subsequent tests. 
+
+Fortunately, the default language can be set in the tearDown method::
+
+	def tearDown(self):
+		translation.activate(settings.LANGUAGE_CODE)
+
```
Comment (carltongibson) [inline docs/topics/testing/tools.txt:696]: Docs should be wrapped at 79 chars. See https://docs.djangoproject.com/en/4.1/internals/contributing/writing-code/coding-style/#python-style

I'd rephrase but put it in a callout, before the `More details...` line. 

```suggestion
.. note::

    When using these methods, ensure to reset the active language at the end of
    each test::

        def tearDown(self):
            translation.deactivate()

More details are in :ref:`how-django-discovers-language-preference`.
```

**Review by carltongibson — COMMENTED** [review]
Hi @fnaimi66 — thanks for this. Looks good. 

Just one comment. 

I adjusted the PR title to match the project [commit message format](https://docs.djangoproject.com/en/4.1/internals/contributing/committing-code/#committing-guidelines).

**Comment by felixxm** [comment]
@fnaimi66 Do you have time to keep working on this?

**Comment by felixxm** [comment]
Updated in #16510.
